File: packages/sdk-py/src/maestro_sdk/connector.py
from typing import Any, Generic, Protocol, TypeVar

from .types import RunContext

TIn = TypeVar("TIn")
TOut = TypeVar("TOut")


class Connector(Protocol, Generic[TIn, TOut]):
    def init(self, ctx: RunContext) -> None: ...  # optional
    def send(self, ctx: RunContext, input: TIn, cfg: dict[str, Any] | None = None) -> TOut: ...


def define_connector(c: Connector[TIn, TOut]) -> Connector[TIn, TOut]:
    return c

File: archive_20250926/_salvage_from_refs/pull/1085/head/packages/sdk-ts/src/connector.ts
import type { RunContext } from './types.js';

export interface ConnectorConfig { [k: string]: unknown }

export interface Connector<TIn = unknown, TOut = unknown> {
  init?: (ctx: RunContext) => Promise<void> | void;
  send: (ctx: RunContext, input: TIn, cfg?: ConnectorConfig) => Promise<TOut>;
}

export function defineConnector<TIn = unknown, TOut = unknown>(c: Connector<TIn, TOut>): Connector<TIn, TOut> {
  return c;
}
